spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/server/services/CloudFoundryInstanceIdGenerator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2014-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.codecentric.boot.admin.server.services;
+
+import de.codecentric.boot.admin.server.domain.values.InstanceId;
+import de.codecentric.boot.admin.server.domain.values.Registration;
+import org.springframework.util.StringUtils;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+/**
+ * Generates InstanceId "cfApplicationGuid:cfInstanceIndex" for CloudFoundry instance
+ * Generates an SHA-1 Hash based on the instance health url for non-CloudFoundry instance
+ */
+public class CloudFoundryInstanceIdGenerator implements InstanceIdGenerator {
+
+    private HashingInstanceUrlIdGenerator hashingInstanceUrlIdGenerator = new HashingInstanceUrlIdGenerator();
```

Comment:
There should be a constructor passing in the a fallbackGenerator. Don't hardwire the hashingInstanceUrlIdGenerator